src/sys/socket/mod.rs

```diff
@@ -487,7 +487,7 @@ cfg_if! {
             pub fn groups(&self) -> &[libc::gid_t] {
                 unsafe {
                     std::slice::from_raw_parts(
-                        self.0.cmcred_groups.as_ptr() as *const libc::gid_t,
+                        self.0.cmcred_groups.as_ptr().cast(),
```

Comment:
This cast seems to be unnecessary? `as_ptr()` already gives us `*const libc::gid_t`

Def: [strcut cmsgcred](https://github.com/rust-lang/libc/blob/6e02a329a2a27f6887ea86952f389ca11e06448c/src/unix/bsd/freebsdlike/mod.rs#L230-L237)

```
    pub struct cmsgcred {
        pub cmcred_pid: ::pid_t,
        pub cmcred_uid: ::uid_t,
        pub cmcred_euid: ::uid_t,
        pub cmcred_gid: ::gid_t,
        pub cmcred_ngroups: ::c_short,
        pub cmcred_groups: [::gid_t; CMGROUP_MAX],
    }
```

Reply:
good catch.